nrf_modem/include/nrf_socket.h

```diff
@@ -909,90 +802,79 @@ int nrf_accept(int sock, void *p_cliaddr, nrf_socklen_t *p_addrlen);
- *          it to a string form of the address.
- *
- * @note    Currently not supporting mixed IPv4 and IPv6 format strings.
+ * @brief Convert IPv4 and IPv6 addresses between binary and text form
  *
- * @param[in]  family  Address family. NRF_AF_INET or NRF_AF_INET6.
- * @param[in]  p_src   Pointer to a struct nrf_in_addr or nrf_in6_addr containing the address to convert.
- * @param[out] p_dst   Pointer to a buffer where the string representation of the address will be stored.
- * @param[in]  size    Size of the provided buffer in p_dst.
- *
- * @return Pointer to p_dst on success, NULL and errno set in case of error.
+ * @details
+ * @rst
+ * See `POSIX.1-2017 article
+ * <https://pubs.opengroup.org/onlinepubs/9699919799/functions/inet_ntop.html>`__
+ * for normative description.
+ * @endrst
  */
-const char *nrf_inet_ntop(int family,
-			  const void *p_src,
-			  char *p_dst,
-			  nrf_socklen_t size);
+const char *nrf_inet_ntop(int af, const void *restrict src, char *restrict dst, nrf_socklen_t size);
 
 /**
- * @brief Function to resolve the host name into IPv4 and/or IPv6 addresses.
- *
- * @note The memory pointed to by @p pp_res must be freed using
- *       nrf_freeaddrinfo when the address is no longer needed
- *       or before calling nrf_getaddrinfo again.
+ * @brief Get address information
  *
- * @param[in]  p_node     Host name to resolve.
- * @param[in]  p_service  Service to resolve.
- * @param[in]  p_hints    Any hints to be used for the resolution.
- * @param[out] pp_res     Pointer to the linked list of resolved addresses if the procedure
- *                        was successful.
+ * @details
+ * @rst
+ * See `POSIX.1-2017 article
+ * <http://pubs.opengroup.org/onlinepubs/9699919799/functions/getaddrinfo.html>`__
+ * for normative description.
+ * @endrst
  *
- * @return 0 if the procedure succeeds.
- *		   Positive eai error on failiure
+ * In addition, the function shall return -1 and set the following errno:
+ * * NRF_ESHUTDOWN If waiting for a modem response while the modem enters a non-initialized state.
  */
-int nrf_getaddrinfo(const char *p_node,
-		    const char *p_service,
-		    const struct nrf_addrinfo *p_hints,
-		    struct nrf_addrinfo **pp_res);
+int nrf_getaddrinfo(const char *restrict nodename,
+		    const char *restrict servname,
+		    const struct nrf_addrinfo *restrict hints,
+		    struct nrf_addrinfo **restrict res);
 
 /**
- * @brief Function for freeing the memory allocated for the result of nrf_getaddrinfo.
- *
- * @details When the linked list of resolved addresses created by nrf_getaddrinfo
- *          is no longer needed, call this function to free the allocated memory.
+ * @brief Free address information returned by @ref nrf_getaddrinfo()
  *
- * @param[in] p_res  Pointer to the memory to be freed.
+ * @details
+ * @rst
+ * See `POSIX.1-2017 article
+ * <http://pubs.opengroup.org/onlinepubs/9699919799/functions/getaddrinfo.html>`__
```

Comment:
```suggestion
 * <http://pubs.opengroup.org/onlinepubs/9699919799/functions/freeaddrinfo.html>`__
```

Reply:
The pages looks to be identical. However, if it should point to getaddrinfo, there should be a note that this is supposed to be getaddrinfo.

Reply:
Updated